src/rpm/builder.rs

```diff
@@ -368,26 +372,145 @@ impl PackageBuilder {
+    ///
+    pub fn pre_untrans_script(mut self, content: impl Into<String>) -> Self {
+        self.pre_untrans_script = Some(PreUntransaction.scriptlet(content.into()));
+        self
+    }
+
+    /// Sets the scriptlet content for `%postuntrans` during package uninstall and upgrade,
+    ///
+    /// **Note** `%postuntrans` scripts are all executed at the end of the transaction that removes their packages
+    ///
+    pub fn post_untrans_script(mut self, content: impl Into<String>) -> Self {
+        self.post_untrans_script = Some(PostUntransaction.scriptlet(content.into()));
         self
     }
 
+    /// Sets the scriptlet content on this builder, interpreting the type from the scriptlet's `ty` field,
+    /// 
+    /// **Overrides** If an existing scriptlet was set, it will be replaced by this fn.
+    /// 
+    /// **Errors** Returns an error if the "ty" field is not set on the Scriptlet type.
+    /// 
+    /// If using one of the below predefined helper pointer structs, this builder method **must** be **infallible**.
+    /// 
+    /// **Example**
+    /// 
+    /// ```rs norun
+    /// builder
+    ///     .scriptlet(PreInstall.scriptlet("echo hello world").flags(..).prog(..))
+    ///     .scriptlet(PostTransaction.scriptlet("echo goodbye world").flags(..).prog(..))
+    /// ```
+    /// 
+    /// **Predefined scriptlet types**
+    /// - `PreInstall`
+    /// - `PostInstall`
+    /// - `PreUninstall`
+    /// - `PostUninstall`
+    /// - `PreTransaction`
+    /// - `PostTransaction`
+    /// - `PreUntransaction`
+    /// - `PostUntransaction`
+    /// 
+    pub fn scriptlet(mut self, scriptlet: impl Into<Scriptlet>) -> Result<Self, crate::Error> {
+        let scriptlet: Scriptlet = scriptlet.into();
+
+        match scriptlet.ty {
+            Some(ty) => match ty {
+                PREIN_TAGS => {
+                    self.pre_inst_script = Some(scriptlet);
+                },
+                POSTIN_TAGS => {
+                    self.post_inst_script = Some(scriptlet);
+                },
+                PREUN_TAGS => {
+                    self.pre_uninst_script = Some(scriptlet);
+                },
+                POSTUN_TAGS => {
+                    self.post_uninst_script = Some(scriptlet);
+                },
+                PRETRANS_TAGS => {
+                    self.pre_trans_script = Some(scriptlet);
+                },
+                POSTTRANS_TAGS => {
+                    self.post_trans_script = Some(scriptlet);
+                },
+                PREUNTRANS_TAGS => {
+                    self.pre_untrans_script = Some(scriptlet);
+                },
+                POSTUNTRANS_TAGS => {
+                    self.post_untrans_script = Some(scriptlet);
+                },
+                _ => {
+                    return Err(Error::NoScriptletTagsSet)
+                }
+            },
+            _ => {
+                return Err(Error::NoScriptletTagsSet)
```

Comment:
This seems like a error that shouldn't need to exist?  Thinking in terms of a package, there cannot be an untyped scriptlet, so why allow one to be created?